app/src/main/java/mollie/tictactoe/MainActivity.java

```diff
@@ -14,14 +14,18 @@
 
 public class MainActivity extends AppCompatActivity {
 
-    private List<Button> mBoard = new ArrayList<>();
+    private final List<Button> mBoard = new ArrayList<>();
     private GameEngine mGameEngine;
+    private static final String EXTRA_GAME_TYPE = "tictactoe.game_type";
+    private static final String EXTRA_BOARD_TYPE = "tictactoe.board_type";
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
-        createGame();
+        String[] players = getIntent().getStringArrayExtra(EXTRA_GAME_TYPE);
```

Comment:
The wording around this intent isn't really clear to me. We are getting something that resembles a game type (from the name) but then assign it to an array of strings that represent players. 

Strings, players, and game types feel incohesive to me. Can we push the conversion of game-type (out of extra) to strings to the actual player instances further out? So this class only ever interacts with players?